gulp/tasks/jekyll-build-preview.js

```diff
@@ -0,0 +1,29 @@
+(function() {
+    'use strict';
+
+    var gulp = require('gulp'),
+        gitUtils = require('../util/gitUtils'),
+        childProcess = require('child_process'),
+        previewConfigFile = '_tmp_preview_config.yml',
+        previewSiteDir = '_preview_site';
+
+    gulp.task('jekyll-build-preview', function() {
+        var branch = gitUtils.currentBranch(),
+            previewBaseUrl = '/' + branch + '/';
+        // Create a temporary Jekyll configuration file which specifies the base URL for the preview site
+        childProcess.execSync(
```

Comment:
Actually, checking the Jekyll documentation, it looks like you can simply pass the base url parameter to jekyll via the `--baseurl` argument, removing the need to spawn a child process to create a temporary file.